Add location list

aviyae opened the pull request:
add location list picker

dhananjaylatkar commented:
LL is stacking on top of each other. Is this expected behavior? 

![image](https://github.com/user-attachments/assets/fb78d744-3af9-4c96-a26d-294c67dcbd0a)

Shouldn't they appear in each window like this -
![image](https://github.com/user-attachments/assets/81c26b27-4af3-4b13-9b72-1d9f665ebf74)

aviyae commented:
Indeed, I changed all the requested changes. Apparently the "botright" option forced the location list to be over all the screen instead of the requested buffer. I fixed that be ignoring it since bottom is the default behaviour (all is documented inside all well)

dhananjaylatkar commented on lua/cscope/init.lua:
```diff
@@ -170,8 +174,12 @@ M.open_picker = function(op_s, symbol, parsed_output)
 	picker_opts.cscope.prompt_title = title
-	picker_opts.cscope.qf_window_size = M.opts.qf_window_size
-	picker_opts.cscope.qf_window_pos = M.opts.qf_window_pos
+	picker_opts.cscope.window_size = M.opts.picker_opts.window_size
```
Please pass the whole opts table instead of deconstructing here.

aviyae replied:
Ok

dhananjaylatkar commented on lua/cscope/pickers/location.lua:
```diff
@@ -0,0 +1,27 @@
+		-- using "botright" like in quickfix will cause several location lists
+		-- opened in the same tab (like in split view) to be stacked one on top of the other
+		-- which is usually not the expected behavior
+		pos = ""
```
I'm assuming you tested other positions and the same problem is not present for them

aviyae replied:
Actually it does :)
But I think that's the only possible way in nvim/vim for location list to be not in the bottom - i.e. I couldn't find a way to locate a location list above (or on the side) without having it being stacked one on top of the other...

dhananjaylatkar replied:
this might help -

[:h :aboveleft](https://neovim.io/doc/user/windows.html#%3Aaboveleft)
[:h :belowright](https://neovim.io/doc/user/windows.html#%3Abelowright)

aviyae replied:
amazing
I totally missed those
will fix
